include/nbl/asset/ICPUDescriptorSet.h

```diff
@@ -28,19 +28,25 @@ namespace nbl::asset
 class NBL_API ICPUDescriptorSet final : public IDescriptorSet<ICPUDescriptorSetLayout>, public IAsset, public impl::IEmulatedDescriptorSet<ICPUDescriptorSetLayout>
 {
 		using impl_t = impl::IEmulatedDescriptorSet<ICPUDescriptorSetLayout>;
+
 	public:
 		using base_t = IDescriptorSet<ICPUDescriptorSetLayout>;
 
 		//! Contructor preallocating memory for SDescriptorBindings which user can fill later (using non-const getDescriptors()).
 		//! @see getDescriptors()
 		ICPUDescriptorSet(core::smart_refctd_ptr<ICPUDescriptorSetLayout>&& _layout) : base_t(std::move(_layout)), IAsset(), impl_t(m_layout.get())
 		{
+			for (uint32_t t = 0u; t < EDT_COUNT; ++t)
+			{
+				if (m_descriptors[t])
+					m_descriptorInfos[t] = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUDescriptorSet::SDescriptorInfo::SBufferImageInfo>>(m_descriptors[t]->size());
+			}
 		}
 
-
 		inline size_t conservativeSizeEstimate() const override
 		{
-			return sizeof(void*)+m_descriptors->size()*sizeof(SDescriptorInfo)+m_bindingInfo->size()*sizeof(impl::IEmulatedDescriptorSet<ICPUDescriptorSetLayout>::SBindingInfo);
+			assert(!"Invalid code path.");
+			return 0xdeadbeefull;
```

Comment:
simple size of the `m_descriptorInfos` and mutable sampler arrays added together and the sizeof the pointer to the layout

Reply:
Are you sure about this? Because you yourself said to return `0xdeadbeef` because this is an unused function.

Reply:
we can leave as is for now